<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1470</width>
    <height>752</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QCustomPlot" name="sound_raw" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>120</y>
      <width>381</width>
      <height>101</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="load_audio">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>116</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Load</string>
    </property>
   </widget>
   <widget class="QPushButton" name="fft">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>30</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>iFFT</string>
    </property>
   </widget>
   <widget class="QCustomPlot" name="sound_raw_fft" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>260</y>
      <width>381</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QCustomPlot" name="sound_raw_fft_phase" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>440</y>
      <width>381</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QCustomPlot" name="sound_filtered" native="true">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>260</y>
      <width>341</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="filter">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>30</y>
      <width>116</width>
      <height>33</height>
     </rect>
    </property>
    <property name="text">
     <string>Filter</string>
    </property>
   </widget>
   <widget class="QCustomPlot" name="sound_inversed_filtered" native="true">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>120</y>
      <width>341</width>
      <height>101</height>
     </rect>
    </property>
   </widget>
   <widget class="QCustomPlot" name="sound_phase_filtered" native="true">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>440</y>
      <width>341</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="image">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>120</y>
      <width>150</width>
      <height>150</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="load_image">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>30</y>
      <width>116</width>
      <height>33</height>
     </rect>
    </property>
    <property name="text">
     <string>Load Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="image_fft_amplitude">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>300</y>
      <width>211</width>
      <height>201</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="image_fft_phase">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>570</y>
      <width>221</width>
      <height>191</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="image_fft_amplitude_filtered">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>300</y>
      <width>191</width>
      <height>211</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="image_fft_filtered">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>120</y>
      <width>150</width>
      <height>150</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="image_fft_phase_filtered">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>570</y>
      <width>231</width>
      <height>191</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="play_sound">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>30</y>
      <width>116</width>
      <height>33</height>
     </rect>
    </property>
    <property name="text">
     <string>Play Original</string>
    </property>
   </widget>
   <widget class="QPushButton" name="play_sound_filtered">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>30</y>
      <width>116</width>
      <height>33</height>
     </rect>
    </property>
    <property name="text">
     <string>Play Filtered</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>90</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Raw Sound</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>90</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Inverse Sound Filtered</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>230</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Raw Sound FFT</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>230</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Filtered Sound</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>410</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>FFT Phase</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>410</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Filtered Phase</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>90</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Original Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>90</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>FFT Filtered Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>280</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>FFT Amplitude</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>280</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>FFT Amplitude Filtered</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>540</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>FFT Phase</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>540</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>FFT Phase Filtered</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="freqSelect">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>70</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100000</number>
    </property>
    <property name="singleStep">
     <number>1000</number>
    </property>
    <property name="value">
     <number>4000</number>
    </property>
   </widget>
   <widget class="QPushButton" name="Stop">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>30</y>
      <width>116</width>
      <height>33</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>70</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Filter Frequency</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="imageFilter">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>40</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="value">
     <number>20</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>10</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Image Filter Frequency</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="HighPass">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>20</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>High Pass</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="LowPass">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>40</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Low Pass</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
